tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java

```diff
@@ -451,6 +452,9 @@ protected void configure() {
 
     private static Module chooseJmapModule(DistributedJamesConfiguration configuration) {
         if (configuration.jmapEnabled()) {
+            if (configuration.searchConfiguration().getImplementation().equals(SearchConfiguration.Implementation.OpenSearch)) {
+                return new TMailJMAPListenerModule();
+            }
```

Comment:
Else we still need to have the compute preview listener?

Reply:
`ComputeMessageFastViewProjectionListener` binding is inside below `JMAPListenerModule`.